### PR TITLE
feat: add requestId to all logs

### PR DESCRIPTION
**Pull request by mweibel**
additionally this enables the use of docker (#32) via traefik.
Wanted to add that since a long time but well, I add it now as I'm
anyway editing the middlewares.. ;)

### CODE REVIEW

File: src/Classes/Utility/MiddlewareForHttpUtility.php [hunk 1]
```diff
@@ -129,5 +132,23 @@ public static function addMiddleware(App $app): void
                 return $response->write('<html lang="en"><head><title>Error</title></head><body><p><strong>Status:</strong>cors  error</p></body>')->withStatus(StatusCode::HTTP_UNAUTHORIZED);
             },
         ]));
+
+        if ('cli-server' == php_sapi_name() && 'DEV' === SITE_ENV) {
```
Comment (cbuchli) [inline src/Classes/Utility/MiddlewareForHttpUtility.php]: you could use ServerUtility::isLocalDevEnv() here.

File: src/Classes/Utility/MiddlewareForHttpUtility.php [hunk 2]
```diff
@@ -129,5 +132,23 @@ public static function addMiddleware(App $app): void
...
+            $app->add(new \RKA\Middleware\ProxyDetection([$_SERVER['REMOTE_ADDR']]));
```
Comment (cbuchli) [inline src/Classes/Utility/MiddlewareForHttpUtility.php]: I'm not entirely sure, but it looks like this Middleware could render ServerUtility::getClientIp() and the isBehindReverseProxy Stuff obsolete? If yes, could you replace it accordingly?

File: src/Classes/Utility/MiddlewareForHttpUtility.php [hunk 3]
```diff
@@ -129,5 +132,23 @@ public static function addMiddleware(App $app): void
...
+        }
+
+        $app->add(function (ServerRequestInterface $request, ResponseInterface $response, callable $next): ResponseInterface {
+            $requestId = $request->getHeader('X-Choria-Request-ID');
+            if (!$requestId) {
+                $requestId = Uuid::uuid4()->toString();
+            }
+            LogHelper::pushProcessorToAllInstances(function (array $record) use ($requestId): array {
```
Comment (cbuchli) [inline src/Classes/Utility/MiddlewareForHttpUtility.php]: Are you sure that no Instance of any Logger gets instantiated after this?

File: src/Classes/Utility/MiddlewareForHttpUtility.php (hunk 2 above)
Comment (mweibel) [reply src/Classes/Utility/MiddlewareForHttpUtility.php]: as discussed: this only rewrites the request URI and doesn't care much about the IP itself

File: src/Classes/Utility/MiddlewareForHttpUtility.php (hunk 3 above)
Comment (mweibel) [reply src/Classes/Utility/MiddlewareForHttpUtility.php]: not really, but as discussed let's leave it for now

**Pull request merged**